src/opnsense/service/templates/OPNsense/Dnsmasq/dnsmasq.conf

```
@@ -59,8 +59,14 @@ bind-interfaces
 bogus-priv
 {% endif %}
 
-{% for override in helpers.toList('dnsmasq.domainoverrides') %}
-server=/{{override.domain}}/{{override.ip}}{%if override.port %}#{{override.port}}{% endif %}{%if override.srcip %}@{{override.srcip}}{% endif %}
+{# Domain Overrides can be sorted by sequence to support strict order #}
+{% set unsorted_domainoverrides = helpers.toList('dnsmasq.domainoverrides') %}
+{% for item in unsorted_domainoverrides %}
```

```suggestion
{% for override in helpers.toList('dnsmasq.domainoverrides', 'sequence') %}
```

is likely all this needs :)

https://github.com/opnsense/core/blob/b7ed45db20d67d1e93fa67d373894bcb6202dd4e/src/opnsense/service/modules/addons/template_helpers.py#L99